website/pages/docs/security/security-models/core.mdx

```diff
@@ -0,0 +1,380 @@
+
+- **mTLS** - Mutual authentication of both the TLS server and client x509 certificates prevents internal abuse through 
+  unauthorized access to Consul agents within the cluster.
+
+  - [`verify_incoming`](/docs/agent/options#verify_incoming) - By default this is false, and should almost always be set
+    to true to require TLS verification for incoming client connections. This applies to both server RPC and to the 
+    HTTPS API.
+
+  - [`verify_incoming_https`](/docs/agent/options#verify_incoming_https) - By default this is false, and should be set 
+    to true to require clients to provide a valid TLS certificate when the Consul HTTPS API is enabled.
+
+  - [`verifing_incoming_rpc`](/docs/agent/options#verify_incoming_rpc) -  By default this is false, and should almost 
+    always be set to true to require clients to provide a valid TLS certificate for Consul agent RPCs.
+
+  - [`verify_outgoing` - By default this is false, and should be set to true to require TLS for outgoing connections from
+    server or client agents. Servers that specify `verify_outgoing = true` will always talk to other servers over TLS, 
+    but they still accept non-TLS connections to allow for a transition of all clients to TLS. Currently the only way to 
+    enforce that no client can communicate with a server unencrypted is to also enable `verify_incoming` which requires 
+    client certificates too.
+
+  - `enable_agent_tls_for_checks`](/docs/agent/options#verify_outgoing) - By default this is false, and should almost 
+    always be set to true to require mTLS to set up the client for HTTP or gRPC health checks. This was added in 
+    Consul 1.0.1.
+
+  - `verify_server_hostname` - By default this is false, and should be set to true to require for outgoing TLS 
+    connections that the TLS certificate presented by the servers matches `server.<datacenter>.<domain> hostname`. The 
+    default configuration does not verify the hostname of the certificate, only that it is signed by a trusted CA. This 
+    setting is critical to prevent a compromised client agent from being restarted as a server and having all cluster 
+    state including all ACL tokens and Connect CA root keys replicated to it, and introduced in 0.5.1. From version 
+    0.5.1 to 1.4.0 we documented that `verify_server_hostname` being true implied verify_outgoing however due to a bug
+    this was not the case so setting only `verify_server_hostname` results in plaintext communication between client 
+    and server. See [CVE-2018-19653](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-19653) for more details. 
+    This is fixed in 1.4.1.
+
+    **Example Server Agent TLS Configuration**
+
+    ```hcl
+    verify_incoming        = true
+    verify_outgoing        = true
+    verify_server_hostname = true
+    
+    ca_file   = "consul-agent-ca.pem"
+    cert_file = "dc1-server-consul-0.pem"
+    key_file  = "dc1-server-consul-0-key.pem"
+    
+    auto_encrypt {
```

Comment:
This is the first time in the document `auto_encrypt` is mentioned, which is odd given how everything else is either already described (`verify_*`) by this point, or is somewhat self-explanatory (`ca_file`).

Reply:
Solid point. I should definitely mention/explain `auto_encrypt`.